### PR TITLE
Add more uv docs

### PR DESCRIPTION
**Pull request by Hoverbear**
Takes a shot at clarifying and improving our `uv` docs.

### CODE REVIEW

**Review by Urhengulas — APPROVED** [review]

**Review by Urhengulas — CHANGES_REQUESTED** [review]
Actually I am unsure if it works for. I did run `setup-venv.sh` which gives:

```sh
$ ./ferrocene/ci/scripts/setup-venv.sh 
+ command -v uv
+ curl -LsSf https://astral.sh/uv/install.sh
+ sh
downloading uv 0.5.1 x86_64-unknown-linux-gnu
no checksums to verify
installing to /home/urhengulas/.local/bin
  uv
  uvx
everything's installed!
+ source /home/urhengulas/.cargo/env
++ case ":${PATH}:" in
+ echo 'source /home/urhengulas/.cargo/env'
./ferrocene/ci/scripts/setup-venv.sh: line 15: BASH_ENV: unbound variable
```
The last line sounds like an error? Assumingly from line 15.

When I then run `uv run which python` I get my system python which is not what happened in your demo today.

```console
$ which python
/usr/bin/python
$ uv run which python
/usr/bin/python
```

When I then run `uv venv ~/.venv` explcitly it actually creates the venv and then which gives the right python:

```console
$ uv venv ~/.venv
Using CPython 3.12.7 interpreter at: /usr/bin/python
Creating virtual environment at: /home/urhengulas/.venv
Activate with: source /home/urhengulas/.venv/bin/activate
$ uv run which python
/home/urhengulas/.venv/bin/python
```

**Comment by Hoverbear** [comment]
Great catch, thanks. :) I'll look into this.

**Comment by Hoverbear** [comment]
@Urhengulas Can I ask you to use the latest main? I think yours was out of date and we only recently made this portable outside CI.

**Comment by Dajamante** [comment]
Thanks for clarifying!
The point that I still find a bit confusing. Do we still need to update our `config. tool` to match CIs?
I am asking because the indentation for this instruction does not make it clear.

File: ferrocene/doc/internal-procedures/src/working-with-the-ci.rst [hunk 1]
```diff
@@ -112,6 +114,15 @@ For example, to run our licensing checks you can run:
    
    uv run reuse --include-submodules lint
 
+Or, to run the ``split-tasks.py`` script for debugging:
+
+.. code-block:: bash
+   
+   uv run ferrocene/ci/split-tasks.py test
+
+In CI workflows you should always use ``uv run`` for calling Python. In the case of ``make.py`` it
```
Comment (Hoverbear) [inline ferrocene/doc/internal-procedures/src/working-with-the-ci.rst:123]:
```suggestion
In CI workflows you should always use ``uv run`` for calling Python. In the case of ``x.py`` it
```

**Comment by Hoverbear** [comment]
> Do we still need to update our config.toml to match CI's when working with uv, aka running CI=true FERROCENE_HOST=aarch64-unknown-linux-gnu ./ferrocene/ci/configure.sh?
I am asking because the indentation for this instruction does not make it clear.

No, I'm not really sure where this comes from?

Are you referring to this separate heading after this one? 

image


They are separate ideas, and unrelated.

**Review by Urhengulas — APPROVED** [review]
With the latest main it works!

**Comment by Dajamante** [comment]
> No, I'm not really sure where this comes from?
> 
> Are you referring to this separate heading after this one?

Yes, I was referring to the separate idea.
Because I never use `uv`, just docker, where I needed to change the `config.toml` to match CI, I didn't know if it was the case for when we use `uv`.

File: ferrocene/doc/internal-procedures/src/working-with-the-ci.rst (hunk 1 above)
Comment (Dajamante) [reply ferrocene/doc/internal-procedures/src/working-with-the-ci.rst]: Doesn't `make` do it as well, by calling its own file of requirements?

**Comment by Hoverbear** [comment]
@Dajamante you do not need to change your config to match CI unless you are trying to reproduce the CI. (Even then, you likely will not use the same config exactly since you likely don't want to have to do AWS auth to sign packages etc)

**Pull request merged**